PC/_msi.c

```diff
@@ -283,6 +283,7 @@ msiobj_dealloc(msiobj* msidb)
 {
     MsiCloseHandle(msidb->h);
     msidb->h = 0;
+    PyObject_Del(msidb);
```

Comment:
I'm not sure that calling PyObject_Del() in dealloc is correct. Maybe you should call Py_TYPE(op)->tp_free(op); instead.

Reply:
From the documentation of tp_dealloc() (https://docs.python.org/3.7/c-api/typeobj.html#c.PyTypeObject.tp_dealloc):
> If the type is not subtypable (doesn’t have the Py_TPFLAGS_BASETYPE flag bit set), it is permissible to call the object deallocator directly instead of via tp_free.

As none of the four types in _msi.c set a tp_free() function, it is correct to call PyObject_Del() here (this is done by many other types).

Reply:
Oh ok. I checked and you are right: range_dealloc() also calls PyObject_Del() for example.

I forgot that PyObject_Del() is simply an alias to PyObject_Free(), a memory deallocator.